common/DbHandler.py

```diff
@@ -1,73 +1,57 @@
 
 from common.Link import Link
 from common.User import User
 
-links = [
-    Link("www.stackoverflow.com", ["programming"]),
-    Link("www.geeksforgeeks.com", ["programming", "learning"])
-]
-
-USERS = {
-    'user1': User(1, 'user1', 'zanjan', links[0]),
-    'test': User(2, 'test', 'test', links)
-}
+from db import db
 
 
 class DbHandler():
+    @staticmethod
+    def get_user_object(username: str):
+        return User.query.filter_by(username=username).first()
+
     @staticmethod
     def validate_login(username: str, password: str):
-        user = USERS.get(username)
+        user = User.query.filter_by(username=username).first()
 
-        return_message = ""
-        if user and safe_str_cmp(
-            user.password.encode('utf-8'), password.encode('utf-8')
-        ):
-            return_message = user
+        # Chech if user exists and password is correct
+        if user and user.check_password(password):
+            return user
         else:
-            return_message = None
-
-        return return_message
+            return None
 
     @staticmethod
     def add_new_user(username: str, password: str):
         # Check if user exists
-        if username in USERS:
+        user = User.query.filter_by(username=username).first()
+        if user:
             return 1
 
-        # If user doesn't exist, get Id for it and Signup
-        max_id = 0
-        for user in USERS.values():
-            if user.id > max_id:
-                max_id = user.id
-
-        # Add new user
-        USERS[username] = User(max_id+1, username, password)
+        user = User(username=username, password_hash=password)
+        db.session.add(user)
+        db.session.commit()
 
         return 0
 
     @staticmethod
-    def get_links(username: str):
-        return USERS.get(username).links
-
-    @staticmethod
-    def append_new_link(username: str, new_link: Link):
-        USERS[username].append_new_link(new_link)
+    def append_new_link(new_link: Link):
+        db.session.add(new_link)
+        db.session.commit()
         return 0
 
     @staticmethod
-    def remove_link(username: str, address_name: str):
-        link_found_status = False
-
-        for (index, link) in enumerate(USERS[username].get_links()):
-            if link.get_address_name() == address_name:
-                link_found_status = True
-                del(USERS[username].get_links()[index])
-                break
-
-        if link_found_status:
-            return 0
+    def remove_link(username: str, link_id: int):
+        user_object = User.query.filter_by(username=username).first()
+        link_object = Link.query.filter_by(id=link_id).first()
+
+        # Check if id exists and it's owner sends request
+        if link_object:
+            if link_object.owner_id == user_object.id:
+                Link.query.filter_by(id=link_id).delete()
+                db.session.commit()
+                return 0
```

Comment:
It would be good to return proper status instead of `0`, `1`, or `2`.

Reply:
Would you mind please do me a favor and clarify what do you mean by `proper status`?
Do you mean to use something like [this](https://docs.python.org/3.4/library/os.html#os.EX_OK) or [raising an exception](https://stackoverflow.com/questions/99683/which-and-why-do-you-prefer-exceptions-or-return-codes)?
([As far as I know](https://stackoverflow.com/questions/35818977/does-python-have-an-exit-success-constant), that would be better to throw exceptions instead of returning numbers as status codes; Alongside that throwing exceptions has its own advantages and dis-advantages)

Reply:
You can raise exceptions or use string (like `OK`, `LINK_DOES_NOT_EXISTS`, ...) instead of a number.